Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+
+`.Rbuildignore` is a way to resolve some of the tension between the practices that support your development process and CRAN's requirements for submission and distribution. The affected files fall into two broad, semi-overlapping classes:
+
+  * Files that help you generate package contents programmatically. Examples:
+    - Using `README.Rmd` to generate an informative and current `README.md`.
+    - Storing `.R` scripts to create and update internal or exported data.
+  * Files that drive package development, checking, and documentation, outside
+    of CRAN's purview. Examples:
+    - Files relating to the RStudio IDE.
+    - Using the [pkgdown package](https://pkgdown.r-lib.org) to generate a
+      website.
+    - Configuration files related to continuous integration/deployment and 
+      monitoring test coverage.
+
+Here is a non-exhaustive list of typical entries in the `.Rbuildignore` file for a package in the tidyverse:
+
+```
+^.*\.Rproj$         # Designates the directory as an RStudio Project
+^\.Rproj\.user$     # Used by RStudio for temporary files
+^README\.Rmd$       # An Rmd file used to generate README.md
+^LICENSE\.md$       # Full text of the license
+^cran-comments\.md$ # Comments for CRAN submission
+^\.travis\.yml$     # Used by Travis-CI for continuous integration testing
+^data-raw$          # Code used to create data included in the package
+^pkgdown$           # Resources used for the package website
+^_pkgdown\.yml$     # Configuration info for the package website
+^\.github$          # Contributing guidelines, CoC, issue templates, etc.
+```
+
+Note that the comments above are not generally present in `.Rbuildignore`; they are included here only for exposition.
+
+We'll mention when you need to add files to `.Rbuildignore` whenever it's important. Remember that `usethis::use_build_ignore()` is an attractive way to manage this file.
+
+### Binary package
+
+If you want to distribute your package to an R user who doesn't have package development tools, you'll need to make a __binary__ package. Like a package bundle, a binary package is a single file. Unlike a bundled package, a binary package is platform specific and there are two basic flavors: Windows and macOS. (Linux users are generally required to have the tools necessary to install from `.tar.gz` files.)
+
+Binary packages for macOS are stored as `.tgz`, whereas Windows binary packages end in `.zip`. If you need to make a binary package, use `devtools::build(binary = TRUE)` on the relevant operating system. Under the hood, this calls `pkgbuild::build(binary = TRUE)` and, ultimately, `R CMD INSTALL --build`, which is described further in the [Building binary packages][wre-binary] secion of [Writing R Extensions][wre].
```

Comment:
Do you think it's that useful to include the devtools code snippets in the text? Most people won't ever call them, so I wonder if they could be relegated to one big diagram at the end of the chapter.

Reply:
I basically do think these are useful, but I take your point. I'm going to leave this here for now for a couple reasons:

  * I think people (myself included) use this book in various ways and one of them is as a reference or for (re-)learning about things you don't do very often.
  * I think it's better accessibility-wise to avoid putting code (only) in a diagram.
  * We can always revisit this later, when/if better or more diagrams are in place.